### ISSUE TITLE
IOS: CBCentralManager implementation of RestoreIdentifier is missing

### ISSUE THREAD

**Issue by mariusbloemhof** [xabre/xamarin-bluetooth-le#204]
Hi,

on IOS - we are missing the Implementation of **RestoreIdentifier** 

```
            _centralManager = new CBCentralManager(this, DispatchQueue.CurrentQueue, new **CBCentralInitOptions
            {
                ShowPowerAlert = true,
                RestoreIdentifier = "myapp.indentifier"
            });**
```
Also **WillRestoreState** is required to make background modes behave correctly on IOS.

The class that contains CBCentralManager will need to inherit from CBCentralManagerDelegate for this to work

**Comment by smsissuechecker** [comment]
Hi @mariusbloemhof,

I'm the friendly issue checker.
It seems like (100.00 %) you haven't used our issue template :cry: I think it is very frustrating for the repository owners, if you ignore them.

If you think it's fine to make an exception, just ignore this message.
**But if you think it was a mistake to delete the template, please close the issue and create a new one.**

Thanks!

**Comment by mariusbloemhof** [comment]
Hi, any feedback about this enhancement?

**Comment by pravincar** [comment]
Hi @smstuebe, I think this is straightforward to implement and is a pretty important feature (for us at least). I'm happy to help if this is not on the radar soon.

**Comment by JanNepras** [comment]
@pravincar I believe that we need to reimplement creating CBCentralManager in [BleImplementation.cs](https://github.com/xabre/xamarin-bluetooth-le/blob/master/Source/Plugin.BLE.iOS/BleImplementation.cs). I am not sure if this is enough according to apple documentation [here](https://developer.apple.com/library/archive/documentation/NetworkingInternetWeb/Conceptual/CoreBluetooth_concepts/CoreBluetoothBackgroundProcessingForIOSApps/PerformingTasksWhileYourAppIsInTheBackground.html) you need implement constructor which have DispatchQueue null and delegete set to self. @xabre @smstuebe With some advices I would be happy to implement it.

**Comment by JanNepras** [comment]
@xabre @smstuebe I found out that this is implemented and merged on branch 2.0.0 [here](https://github.com/xabre/xamarin-bluetooth-le/pull/314). Would it be possible to release it?

**Comment by xabre** [comment]
Release with v2.0.1
